db/db_impl/db_impl.cc

```diff
@@ -4890,26 +4910,28 @@ Status DBImpl::VerifyChecksumInternal(const ReadOptions& read_options,
   return s;
 }
 
-Status DBImpl::VerifySstFileChecksum(const FileMetaData& fmeta,
+Status DBImpl::VerifySstFileChecksum(const std::string& file_checksum_expected,
+                                     const std::string& func_name_expected,
                                      const std::string& fname,
                                      const ReadOptions& read_options) {
   Status s;
-  if (fmeta.file_checksum == kUnknownFileChecksum) {
+  if (file_checksum_expected == kUnknownFileChecksum) {
```

Comment:
I think we should also bail out here if `file_checksum_expected` is empty.

Reply:
```kUnknownFileChecksum``` is intialized to empty string.